pex/resolve/lock_resolver.py

```diff
@@ -117,7 +117,9 @@ def resolve_from_lock(
         )
         parsed_requirements = try_(
             parse_lockable_requirements(
-                requirement_configuration, network_configuration=network_configuration
+                requirement_configuration,
+                network_configuration=network_configuration,
+                fallback_requirements=(str(req) for req in lock.requirements),
```

Comment:
`Lockfile.requirements` is a sequence of `pkg_resources.Requirement`, whereas we need `pex.requirements.ParsedRequirement`. Our parser for that, `parse_requirement_strings`, expects strings.